support/vm/invocation_context.go

```diff
@@ -31,6 +31,23 @@ import (
 
 var EmptyObjectCid cid.Cid
 
+type StatsByCall map[MethodKey]*CallStats
+
+type MethodKey struct {
+	Code   cid.Cid
+	Method abi.MethodNum
+}
+
+type CallStats struct {
+	Reads    uint64
+	Puts     uint64
+	Calls    uint64
+	SubStats StatsByCall
+
+	startReads uint64
+	startPuts  uint64
+}
```

Comment:
Break this out from this large file, and so that the methods can be placed adjacent. Please document how `startReads` and `startPuts` are to be used.

`StatsByCall` too.